homeassistant/components/haiku/fan.py

```diff
@@ -0,0 +1,170 @@
+        self._supported_features = SUPPORT_SET_SPEED
+        self._speed_list = [
+            SPEED_OFF,
+            SPEED_1,
+            SPEED_2,
+            SPEED_3,
+            SPEED_4,
+            SPEED_5,
+            SPEED_6,
+            SPEED_7,
+        ]
+        self._speed = SPEED_1
+        self._state = SPEED_1
+        self._entities = self._entity_id
+
+    @property
+    def name(self):
+        """Return the display name of this fan."""
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        """Set the name of the fan."""
+        self._name = name
+
+    @property
+    def unique_id(self):
+        """Return the display name of this fan."""
+        return self._unique_id
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self._supported_features
+
+    @property
+    def speed_list(self) -> list:
+        """Get the list of available speeds."""
+        return self._speed_list
+
+    @property
+    def capability_attributes(self):
+        """Return capability attributes."""
+        return {ATTR_SPEED_LIST: self.speed_list}
+
+    @property
+    def state_attributes(self) -> dict:
+        """Return optional state attributes."""
+        data = {}
+        data[ATTR_SPEED] = self.speed
+        return data
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state != STATE_OFF
+
+    @property
+    def state(self):
+        """Return state of device."""
+        return self._state
+
+    @property
+    def speed(self):
+        """Return the current speed."""
+        if self._speed in self._speed_list:
+            self._speed = self._speed
+            self._state = self._state
+        else:
+            self._speed = SPEED_OFF
+            self._state = STATE_OFF
+        return self._speed
+
+    async def async_update(self):
+        """Update state of device."""
+        updatestate = await fan.getspeed(self._cip)
+        if updatestate is None or updatestate is False:
+            self._state = STATE_OFF
+            self._speed = SPEED_OFF
+        elif updatestate == "0":
+            self._state = STATE_OFF
+            self._speed = SPEED_OFF
+        else:
+            self._speed = self._speed_list[updatestate]
+            self._state = self._speed_list[updatestate]
+
+    async def async_set_speed(self, speed: str) -> None:
+        """Set the speed of the fan."""
+        if speed in self._speed_list:
+            hcspeed = (speed.split(" "))[1]
+            await fan.setspeed(self._cip, ("" + str(hcspeed)))
+            self._speed = speed
+            self._state = speed
+        else:
+            _LOGGER.error(
+                "Received invalid speed: %s. Expected: %s.", speed, self._speed_list
+            )
+
+    async def async_turn_on(self, speed: str = None) -> None:
+        """Turn on the fan."""
+        if speed == "off":
```

Comment:
Use the speed constants available.
```suggestion
        if speed == SPEED_OFF:
```

Reply:
Okay.